### ISSUE TITLE
Decide how we'll do deployments per client

### ISSUE THREAD

**Issue by osdiab** [osdiab/spinach-web-app#4]
One option: different clients, different instances/tenants of vercel + auth0 + sendgrid + whatever other third parties we use. need a way to keep track of all the different stuff that's different per deploy.

We don't need to worry about this for a while, for now we only have one client, but something to design around

**Comment by AndyMoreland** [comment]
Email is interesting because using the intermediary's domain is non-trivial from an integration pov. We'd have to get spf + other security stuff set up properly with their IT admins, I think.

**Comment by osdiab** [comment]
Yeah, also some nuance around email credibility/getting caught by spam filters - probably separate warm-up phases per custom domain. once again, something to keep in mind but not something we need to worry about short term, just whatever we do ought to be done in a way that isn't a pain to configure later.

Omar

**Comment by osdiab** [comment]
for now I don't think we need to be particularly automated about it. I'm thinking we can use [Sendgrid subusers](https://sendgrid.com/docs/ui/account-and-settings/subusers/) for this, make one per intermediary and register a notifications@customdomain.com sender address. We'd work with their IT admins to add the proper domain stuff. In the longer term we can also write a script to poll DNS records, remind them if they haven't done it right.

Long-term Sendgrid has [API endpoints for subusers](https://sendgrid.com/docs/for-developers/sending-email/how-to-create-a-subuser-with-the-api/) so we can probably use that to do this process self-serve, but I don't see the volume of intermediaries coming in to be so high that that would be a major issue for a while.

There's also the question of deliverability/sender reputation. For now we can roll with just using the shared IPs that Sendgrid provides, or at most getting a single dedicated IP for all of our properties. Since we're only using this for notification purposes on individual clients and not for email campaigns per se, I'm not too worried about meticulously mantaining a high sender reputation/being super careful about warming up IPs and stuff. 

That all sound good?

**Comment by osdiab** [comment]
A requirement here for automating the process of spinning up a new intermediary, is to be able to duplicate all the templates for a new subuser. In order to do that we'd need to use the transactional templates API - for each template in the top account, make a duplicate one for the subuser, and then for the latest version of that template, fetch it and make an identical template for the newly created subuser template. Template ID can be specified, so we can always match the same template ID as the parent.

If we can handle the theming with parameters, we can do it that way; or we can edit the templates somehow in this copy process, but that feels brittle.

I figure we can do this as a script, which would be sufficient. would wanna make sure that failures are clearly pointed out so that we can patch the gaps if there are any.

**Comment by osdiab** [comment]
For now I'm not messing with subusers because it requires a Pro plan and we only have one starter intermediary.
